Content.Server/Resist/ResistLockerSystem.cs

```diff
@@ -0,0 +1,124 @@
+using Content.Shared.Movement;
+using Robust.Shared.GameObjects;
+using Content.Server.Storage.Components;
+using Content.Server.DoAfter;
+using Robust.Shared.IoC;
+using Robust.Shared.Player;
+using Robust.Shared.Containers;
+using Content.Server.Popups;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Resist
+{
+    public class ResistLockerSystem : EntitySystem
+    {
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ResistLockerComponent, RelayMovementEntityEvent>(OnRelayMovement);
+            SubscribeLocalEvent<ResistDoAfterComplete>(OnDoAfterComplete);
+            SubscribeLocalEvent<ResistDoAfterCancelled>(OnDoAfterCancelled);
+            SubscribeLocalEvent<ResistLockerComponent, EntRemovedFromContainerMessage>(OnRemovedFromContainer);
+        }
+
+
+        private void OnRelayMovement(EntityUid uid, ResistLockerComponent component, RelayMovementEntityEvent args)
+        {
+            TryComp(uid, out EntityStorageComponent? storageComponent);
+            if (!Resolve(uid, ref storageComponent))
+                return;
+
+            if (!component.IsResisting)
+            {
+                if (TryComp<LockComponent>(uid, out var lockComponent) && lockComponent.Locked || storageComponent.IsWeldedShut)
+                {
+                    AttemptResist(args.Entity, uid, storageComponent, component);
+                } 
+            }
+
+        }
+        private void AttemptResist(EntityUid user, EntityUid target, EntityStorageComponent? storageComponent, ResistLockerComponent? resistLockerComponent)
+        {
+            if (!Resolve(target, ref storageComponent, ref resistLockerComponent))
+                return;
+
+            resistLockerComponent.CancelToken = new();
+            var doAfterEventArgs = new DoAfterEventArgs(user, resistLockerComponent.ResistTime, resistLockerComponent.CancelToken.Token, target)
+            {
+                BreakOnTargetMove = false,
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = false, //No hands 'cause we be kickin'
+                BroadcastFinishedEvent = new ResistDoAfterComplete(storageComponent, resistLockerComponent, user, target),
+                BroadcastCancelledEvent = new ResistDoAfterCancelled(user, resistLockerComponent)
```

Comment:
You can probably use `TargetFinishedEvent` and `TargetCancelledEvent` here instead to have:
```
SubscribeLocalEvent<ResistLockerComponent, ResistDoAfterComplete>(OnDoAfterComplete);
SubscribeLocalEvent<ResistLockerComponent, ResistDoAfterCancelled>(OnDoAfterCancelled);
```

Reply:
That's a great point, will do, thanks!